packages/core/src/util/tokenTransferInspector.ts

```diff
@@ -118,5 +160,8 @@ export const tokenTransferInspector: TokenTransferInspector =
     removeZeroBalanceEntries(fromAddress);
     removeZeroBalanceEntries(toAddress);
 
-    return { fromAddress, toAddress };
+    return {
+      fromAddress: await toTokenTransferValue(assetProvider, fromAddress),
```

Comment:
It would be great to **not** look up cip68 metadata in `fromAddress`: there will be dapps that makes transactions to 'mutate' metadata, so ideally wallet would show old metadata in 'fromAddress' and new metadata in 'toAddress'

Reply:
Ok so basically, in the fromAddress we do not want to check the transaction for metadata, and only rely on cache and backend correct?

Reply:
Correct

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1076/commits/f16e095a0f51587495ca35b81ff2538b4baab671